clients/da-vinci-client/src/main/java/com/linkedin/davinci/client/AvroGenericDaVinciClient.java

```diff
@@ -451,8 +422,7 @@ public void compute(
       Schema computeResultSchema = getComputeResultSchema(computeRequestWrapper);
 
       for (K key: keys) {
-        byte[] keyBytes =
-            keySerializer.serialize(key, reusableObjects.binaryEncoder, reusableObjects.byteArrayOutputStream);
+        byte[] keyBytes = keySerializer.serialize(key);
```

Comment:
No need to do this in the current PR, but it would be interesting to try to:

1. Use Avro's `ByteBufferOutputStream` to serialize into a threadlocal `ByteBuffer`.
2. Then use RocksDB's `get(final ReadOptions opt, final byte[] key, final int offset, final int len, final byte[] value, final int vOffset, final int vLen)` in order to have reuse on both the passed in key and the returned value.

Reply:
I mean that the key serializer will instantiate a new `byte[]`, from my understanding, but even that part could be reused I think, if we made the serializer populate a threadlocal `ByteBuffer`.

It would require a somewhat big refactoring though, so we can drop it.

Reply:
You're right, `ByteArrayOutputStream.toByteArray` makes deep copy of its content. Thanks for explaining.